cfme/cloud/provider/azure.py

```diff
@@ -27,7 +28,14 @@ def _form_mapping(self, create=None, **kwargs):
     def configloader(cls, prov_config, prov_key):
         credentials_key = prov_config['credentials']
         credentials = cls.process_credential_yaml_key(credentials_key)
-        return cls(name=prov_config['name'],
+        # HACK: stray domain entry in credentials, so ensure it is not there
+        credentials.domain = None
+        region = prov_config.get('region', None)
+        if region is not None and isinstance(region, dict):
+            region = pick(region)
+        return cls(
+            name=prov_config['name'],
+            region=region,
```

Comment:
@jteehan Here I did not bother with verpicking in the code, but rather keeping the values in the yaml. We can't verpick all the values in the yaml, but with region we know there can be differences. In the yaml, you don't have to use the dictionary, as you see here, it is optional. And it also passes the region so it then subsequently gets inserted in the UI.